Source/bmalloc/libpas/src/libpas/pas_thread_local_cache_layout.c

```diff
@@ -71,6 +76,36 @@ pas_allocator_index pas_thread_local_cache_layout_add_node(pas_thread_local_cach
 
     result = pas_thread_local_cache_layout_next_allocator_index;
 
+    did_overflow = __builtin_add_overflow(
+        pas_thread_local_cache_layout_next_allocator_index,
+        pas_thread_local_cache_layout_node_num_allocator_indices(node),
+        &pas_thread_local_cache_layout_next_allocator_index);
+    PAS_ASSERT(!did_overflow);
+
+    begin_page = pas_round_up_to_power_of_2(pas_thread_local_cache_offset_of_allocator(result), page_size);
+    end_page = pas_round_up_to_power_of_2(pas_thread_local_cache_offset_of_allocator(pas_thread_local_cache_layout_next_allocator_index - 1), page_size);
```

Comment:
I would like to ensure that we will never have a local-allocator which exceeds the page size. If it is true, then this is safe.